src/rdiff-backup-delete

```diff
@@ -225,23 +226,45 @@ def _remove_increments(path, dry_run):
 
 def _rmtree(path, dry_run):
     """
-    Custom implementation of shutil.rmtree() to support deletion of symlink.
+    Custom implementation of shutil.rmtree() to handle permission errors,
+    fifo and symlink deletion.
     """
+    if dry_run:
+        return
+
+    # Try to change the permissions of the file or directory to delete them.
+    def onerror(func, path, exc_info):
+        if exc_info[0] == PermissionError:
+            # Parent directory must allow rwx
+            if not os.access(os.path.dirname(path), os.W_OK | os.R_OK | os.X_OK):
+                os.chmod(os.path.dirname(path), 0o0700)
```

Comment:
This would be less intrusive (the owner might want the group to still be able to read the directory):

```python
parent_dir = os.path.dirname(path)
st = os.stat(parent_dir)
os.chmod(parent_dir, st.st_mode | stat.S_IRWXU)
```
You could create a small helper function to do it, because you would need to do something similar for the file itself.

Reply:
Not sure it's required to be done for the file. It's gonna get deleted.